source/common/http/header_map_impl.cc

```diff
@@ -650,5 +658,49 @@ HeaderMapImplUtility::getAllHeaderMapImplInfo() {
   return ret;
 }
 
+absl::optional<absl::string_view>
+RequestHeaderMapImpl::getTraceContext(absl::string_view key) const {
+  ASSERT(validatedLowerCaseString(key));
+  auto result = const_cast<RequestHeaderMapImpl*>(this)->getExisting(key);
+
+  if (result.empty()) {
+    return absl::nullopt;
+  }
+  return result[0]->value().getStringView();
+}
+
+void RequestHeaderMapImpl::setTraceContext(absl::string_view key, absl::string_view val) {
+  ASSERT(validatedLowerCaseString(key));
+  HeaderMapImpl::removeExisting(key);
+
+  HeaderString new_key;
+  new_key.setCopy(key);
+  HeaderString new_val;
+  new_val.setCopy(val);
+
+  HeaderMapImpl::insertByKey(std::move(new_key), std::move(new_val));
```

Comment:
no need to move string_view, it does nothing.

Reply:
🤣 Type of `new_key/new_val` is `Http::HeaderString` rather then string_view and `HeaderMapImpl::insertByKey` use `Http::HeaderString&&` as parameter type. It seems that this move is necessary.

```
void HeaderMapImpl::insertByKey(HeaderString&& key, HeaderString&& value) {
  // ......
}

```